sympy/polys/rootoftools.py

```diff
@@ -394,6 +395,25 @@ def free_symbols(self):
         # symbols
         return set()
 
+    def _eval_power(self, other):
+        if not isinstance(other, (Integer, int)):
+            return None
+        if other == 0:
+            return S.One
+        if other == 1:
+            return self
+        if abs(other) < self.poly.degree():
+            return None
+        if 0 < other:
+            x = self.poly.gen
+        elif other < 0:
+            other = -other
+            try:
+                x = invert(self.poly.gen, self.poly)
+            except NotInvertible:
+                return None
+        return (x**other).as_poly(domain=QQ).rem(self.poly)(self)
```

Comment:
I would rewrite this as
```python
x = Poly([0, 1], self.poly.gen, domain=QQ)
n = other.p
if n > 0:
    p = (x ** n).rem(self.poly)
else:
    p = (x ** -n).invert(self.poly)
return p(self)
```
There are a few ways that `Poly` should make this easier like firstly there should be something analogous to `Poly.gen` but to get the result as a Poly. Secondly Poly should have a method like `powmod` for computing `(a ** n) % b` that can also handle negative powers (maybe even `pow(a, n, b)` should do this.

Reply:
I think using `x = Poly({abs(n): 1}, self.poly.gen, domain=QQ)` might also be fine, but is this not recommended?

Reply:
Yes, that works. The main point is to avoid converting between Poly and Expr as much as possible and also to avoid making the Poly code guess the generators and domain.